include/curves/bezier_curve.h

```diff
@@ -140,6 +140,52 @@ struct bezier_curve : public curve_abc<Time, Numeric, Safe, Point> {
     }
   }
 
+  virtual bool isApprox(const bezier_curve_t& other, const Numeric prec = Eigen::NumTraits<Numeric>::dummy_precision(),const size_t order = 5) const{
+    //std::cout<<"is approx in bezier called."<<std::endl;
+    (void)order;
+    bool equal = T_min_ == other.min()
+        && T_max_ == other.max()
+        && dim_ == other.dim()
+        && degree_ == other.degree()
+        && size_ == other.size_
+        && mult_T_ == other.mult_T_
+        && bernstein_ == other.bernstein_;
+    if(!equal)
+      return false;
+    for (size_t i = 0 ;i < size_;++i)
+    {
+      if(!control_points_.at(i).isApprox(other.control_points_.at(i),prec))
+        return false;
+    }
+    return true;
+  }
+
+
+  virtual bool operator==(const bezier_curve_t& other) const {
+    return isApprox(other);
+  }
+
+  virtual bool operator!=(const bezier_curve_t& other) const {
+    return !(*this == other);
+  }
+
+  virtual bool isApprox(const curve_abc_t& other, const Numeric prec = Eigen::NumTraits<Numeric>::dummy_precision(),const size_t order = 5) const{

```

Comment:
See this post. You should only compare instances of the same class. 
https://stackoverflow.com/questions/4669966/abstract-class-and-operator-in-c 
But then I guess this is again related to the python bindings